libs/datamodel/core/src/transform/ast_to_dml/db/attributes.rs

```diff
@@ -360,6 +356,44 @@ fn primary_key_constraint_name<'ast>(
     db_name
 }
 
+fn default_value_constraint_name<'ast>(
+    args: &mut Arguments<'ast>,
+    ast_model: &'ast Model,
+    model_data: &mut ModelData<'ast>,
+    ast_field: &'ast ast::Field,
+    ctx: &mut Context<'ast>,
+) -> Option<String> {
+    if !ctx.db.preview_features.contains(NamedConstraints) {
+        return None;
+    }
+
+    let db_name = match args.optional_arg("map").map(|name| name.as_str()) {
+        Some(Ok("")) => error_on_empty_string_cow(args, ctx),
+        Some(Ok(name)) => Some(name.into()),
+        Some(Err(err)) => push_error_cow(err, ctx),
+        None if ctx.db.active_connector().supports_named_default_values() => {
+            let generated_name = ConstraintNames::default_name(
+                model_data.mapped_name.unwrap_or(&ast_model.name.name),
+                ast_field.name(),
+                ctx.db.active_connector(),
+            );
+
+            Some(generated_name.into())
+        }
+        None => None,
+    };
+
+    validate_db_name(&ast_model, args, &db_name, "@default", ctx);
+
+    if db_name.is_some() && !ctx.db.active_connector().supports_named_default_values() {
+        ctx.push_error(args.new_attribute_validation_error(
+            "You defined a database name for the default value of a field on the model. This is not supported by the provider.",
+        ));
+    }
+
+    db_name.map(|cow| cow.to_string())
+}
```

Comment:
I already had concerns and re-read the code for constraint names validation in `attributes.rs`, now I'm certain we should change the approach asap, we're going down the wrong path (this code does not belong here). Do you have time for a call today so we can sketch the better approach? I'd also be fine merging this and correcting later, but it's only getting more expensive to fix.

Reply:
We'll have a call later. Let's do this!

Reply:
On a call, we as a team decided to redo this after GA.